services/brig/src/Brig/Provider/DB.hs

```diff
@@ -486,6 +479,26 @@ updateServiceTags pid sid (oldName, oldTags) (newName, newTags)
 -- Used both by service_tag and service_prefix
 type IndexRow = (Name, ProviderId, ServiceId)
 
+-- Note [buggy pagination]
+-- ~~~~~~~~~~~~~~~~
+--
+-- I'm very much unsure that pagination is implemented correctly
+-- in 'paginateServiceNames' and 'paginateServiceTags'.
+--
+-- It's not obvious that it's enough to pass (size+1) and then try to figure
+-- out, in various places, whether we should flip 'hasMore' or not; and if
+-- this is implemented incorrectly, then hasMore might be false while it
+-- should be true – which will become a significant bug once clients
+-- actually start implementing pagination.
+--
+-- In addition, we certainly return less rows than we are asked for (though
+-- it's not exactly a bug), and we can even return zero rows and say "but we
+-- have more!".
+--
+-- Luckily, clients never look at hasMore. There are also some tests that
+-- break if they look at hasMore, but since hasMore is currently irrelevant,
+-- they're commented out. Grep for references to this note.
```

Comment:
> this looks like another case of dead code sticking around

The part about returning `has_more` can be considered dead code because clients don't inspect it. However, we'll have to implement this in the future anyway, and what we currently have is useful enough that I'd rather leave it in